packages/graphql/src/auth/get-jwt.ts

```diff
@@ -100,28 +100,27 @@ async function getJWT(context: Context): Promise<any> {
 }
 
 // Verifies the JWKS asynchronously, returns Promise
-async function verifyJWKS(client: JwksClient, token: string) {
-    function getKey(header, callback) {
+async function verifyJWKS(client: JwksClient, token: string): Promise<JwtPayload | undefined> {
+    function getKey(header: JwtHeader, callback: SigningKeyCallback) {
         // Callback that returns the key the corresponding key[kid]
-        client.getSigningKey(header.kid, (err, key) => {
+        client.getSigningKey(header.kid, (_err, key) => {
             const signingKey = key?.getPublicKey();
             callback(null, signingKey);
         });
     }
 
     // Returns a Promise with verification result or error
-    return new Promise((resolve, reject) =>
+    return new Promise((resolve, reject) => {
         jsonwebtoken.verify(
             token,
             getKey,
             {
                 algorithms: ["HS256", "RS256"],
             },
             (err, decoded) => {
-                return err ? reject(err) : resolve(decoded);
+                if (err) reject(err);
+                else resolve(decoded);
             }
-        )
-    );
+        );
+    });
```

Comment:
Is this right? Having an arrow function with curly braces but no return value doesn't quite seem correct to me.

Reply:
This is actually the fix to an eslint rule of a promise returning. within the promise callback, the return value is irrelevant, as the key is the resolve/reject callbacks

Reply:
So that ESLint rule would have been referring to the return on line 121 (`return resolve(decoded)`) for example. I'm talking about line 113/114 - shouldn't it be `return jsonwebtoken.verify(`?

Reply:
No, the callback inside a Promise (`new Promise...`) does not return anything, the key part is the resolve/reject that are called within the `.verify` callback. Eslint complains about that one, keep in mind that the callback inside .verify is not really the promise callback, is a plain one

Reply:
Interesting! Thanks for the background. 🙂